src/Generators/Generic.php

```diff
@@ -30,6 +30,10 @@
      */
     public function generate()
     {
+        if ($this->kind === false) {
```

Comment:
This still bothers me, since `$this->kind` has doc that says it's a string only.